src/ControlSurface/Controls/ImpulseButton/ScoreImpulseButtons.qml

```diff
@@ -0,0 +1,73 @@
+import QtQuick 2.0
+import QtQuick.Layouts 1.12
+import QtQuick.Controls.Styles 1.4
+import QtQuick.Controls 2.15
+import QtQml.Models 2.12
+
+Flow {
+    spacing: 5
+    width: window.width - 10
+
+    Repeater {
+        id: impulseButtonList
+        clip: true
+        Layout.fillHeight: parent.height
+        Layout.fillWidth: parent.height
+        Layout.margins: 5
+
+        model: ListModel {
+            id: impulseButtonlistModel
+        }
+
+        delegate: ScoreImpulseButton {
+            id: button
+            controlCustom: _custom
+            controlId: _id
+            controlUuid: _uuid
+            controlSurfacePath: path
+
+            isPressed: _isPressed
+        }
+    }
+
+    // Receving informations about impulse buttons in control surface from score
+    Connections {
+        // Adding an impluse button in the control surface
+        function onAppendImpulseButton(s) {
+            function find(cond) {
```

Comment:
I think that this "find" function could go in an utility.js file

Reply:
I am working on it